src/edu/csus/ecs/pc2/core/InternalController.java

```diff
@@ -2659,7 +2661,80 @@ public void sendToTeams(Packet packet) {
             sendPacketToClients(packet, ClientType.Type.TEAM);
         }
     }
+    
+    public void sendToGroupsandIndividualTeams(Packet packet,IElementObject[] ultimateDestination) {
+
+//        Properties properties = (Properties) packet.getContent();
+//        // does the packet includes problemDataFiles
+//        boolean abort = true;
+//        if (properties.containsKey(PacketFactory.PROBLEM_DATA_FILES)) {
+//            // clone before start mucking with it, or do we need a deep clone?
+//            Properties cloneProperties = new Properties();
+//            for (Iterator<?> iter = properties.keySet().iterator(); iter.hasNext();) {
+//                String element = (String) iter.next();
+//                // skip PROBLEM_DATA_FILES, otherwise clone the element
+//                if (!element.equals(PacketFactory.PROBLEM_DATA_FILES)) {
+//                    cloneProperties.put(element, properties.get(element));
+//                    abort = false;
+//                }
+//            }
+//            packet = PacketFactory.clonePacket(packet.getSourceId(), packet.getDestinationId(), packet);
+//            // stick it back into the packet
+//            packet.setContent(cloneProperties);
+//        } else {
+//            abort = false;
+//        }
+//        if (!abort) {
+//            sendPacketToClients(packet, ClientType.Type.TEAM);
+//        }
+        
+        // sendToClient(ConnectionHandlerID connectionHandlerID, Packet packet
+        ClientId[] clientIds = contest.getLocalLoggedInClients(ClientType.Type.TEAM);
+        
+        List<ClientId> badClients = new ArrayList<ClientId>();
 
+        HashSet<ElementId> setWithElementId = new HashSet<>();
+        for (IElementObject element : ultimateDestination) {
+            setWithElementId.add(element.getElementId());
+        }
+        for (ClientId clientId : clientIds) {
+            ConnectionHandlerID connectionHandlerID = null;
+            
+            //For individual accounts
+            Account account = contest.getAccount(clientId);
+   
+            if (setWithElementId.contains(account.getElementId())) {
+                connectionHandlerID = clientId.getConnectionHandlerID();
+                try {
+                    sendToClient(connectionHandlerID, packet);
+                }catch (Exception e) {
+                    getLog().log(Level.WARNING, "Exception attempting to send packet using individual team to client " + clientId + "at connectionHandlerId " + connectionHandlerID
+                            + ": " + packet + ": "+ e.getMessage(), e);
+                
+                }
+                continue;
+            }
+            for (IElementObject destination : ultimateDestination) {
+                if (destination instanceof Account) {
```

Comment:
From what I can tell in SubmitClarification.Submit(), ultimateDestination can only be a list of group element Ids.  Is that right?   If that is the case, then perhaps a refactor is in order to change the name of ultimateDestination to something like destinationGroups ?

Reply:
removed ultimateDestination, it now is done by two different arrays one for groups one for teams